src/clj/rems/db/api_key.clj

```diff
@@ -1,10 +1,67 @@
 (ns rems.db.api-key
-  (:require [rems.db.core :as db]
-            [rems.json :as json]))
+  (:require [clojure.test :refer :all]
+            [rems.db.core :as db]
+            [rems.json :as json]
+            [rems.util :refer [update-present]]))
 
-(defn valid? [key]
-  (not (nil? (db/get-api-key {:apikey key}))))
+(defn- format-api-key [key]
+  (-> key
+      (update-present :users json/parse-string)
+      (update-present :paths json/parse-string)))
 
-(defn add-api-key! [key comment]
+(defn get-api-key [key]
+  (format-api-key (db/get-api-key {:apikey key})))
+
+(defn get-api-keys []
+  (mapv format-api-key (db/get-api-keys {})))
+
+(defn- method= [method pattern]
+  (let [normalized (.toLowerCase (name method))
+        normalized-pattern (.toLowerCase (name pattern))]
```

Comment:
I wonder if it would be better to have them normalized already in db. Could remove some of this code. Also maybe I would use `str` namespace even though here obviously we are in JVM.

Reply:
that's an ok idea, I just wasn't sure if ring normalized the methods or not. I guess I'll normalize everything to lower case.

Reply:
done

Reply:
Instead of normalization I just added validation for the method names given to `add-api-key!`. This has the added benefit of catching typos.